offchain-modules/src/packages/handlers/btc.ts

```diff
@@ -16,12 +17,16 @@ export class BtcHandler {
     logger.debug('start btc watchLockEvents');
     while (true) {
       try {
+        await asyncSleep(1000 * 10);
         const latestHeight = await this.db.getLatestHeight();
-        const nowTips = await this.btcChain.getBtcHeight();
-        logger.debug(`'btc db lock record latest height: ${latestHeight} chain now height: ${nowTips}`);
+        const targetHeight = (await this.btcChain.getBtcHeight()) - ForceBridgeCore.config.btc.confirmNumber;
+        if (targetHeight <= latestHeight) {
+          continue;
+        }
+        logger.debug(`'btc db lock record latest height: ${latestHeight}. target height: ${targetHeight}`);
         await this.btcChain.watchBtcTxEvents(
           latestHeight,
-          nowTips,
+          targetHeight,
```

Comment:
This algorithm is inefficient, because of btcChain will request the same block time and time again, when there's not cross bridge tx happen.